src/transformStateWithClones.js

```
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const arrWithStates = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newState = { ...newState, ...action.extraData };
+        arrWithStates.push(newState);
+        break;
+
+      case 'removeProperties':
+        newState = { ...newState };
+
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        arrWithStates.push(newState);
+        break;
+
+      case 'clear':
+        newState = {};
+        arrWithStates.push(newState);
+    }
```

The switch statement has no default case. This violates checklist item #3: "3. [CODE STYLE]: switch/case should always have default case for error handling." Without a default, unknown action.type values are silently ignored and no state is pushed for that action, breaking the requirement that the returned history contains a state after each action (history[i] must correspond to actions[i]) fileciteturn0file2. Add a default case that either throws for unknown action types or treats them as a no-op but still results in a pushed snapshot.